tests/unittests/Foundation/NSDecimalNumberTests.mm

```diff
@@ -0,0 +1,356 @@
+}
+
+TEST(NSDecimalNumber, InitWithUnsignedLongLong) {
+    unsigned short mantissa[] = { 62556, 31703, 2147, 0, 0, 0, 0, 0, 0 };
+    NSDecimalNumber* dNumber = [[[NSDecimalNumber alloc] initWithUnsignedLongLong:9223372534876] autorelease];
+    NSDecimal decimal = [dNumber decimalValue];
+
+    EXPECT_EQ(3, decimal._length);
+    EXPECT_EQ(0, decimal._exponent);
+    EXPECT_EQ(0, decimal._isNegative);
+
+    EXPECT_ARRAY_EQUAL(mantissa, decimal._mantissa);
+}
+
+TEST(NSDecimalNumber, InitWithDouble) {
+    unsigned short mantissa[] = { 53248, 62560, 115, 32768, 0, 0, 0, 0, 0 };
+    NSDecimalNumber* dNumber = [[[NSDecimalNumber alloc] initWithDouble:9223372534876e24] autorelease];
+    NSDecimal decimal = [dNumber decimalValue];
+
+    EXPECT_EQ(4, decimal._length);
+    EXPECT_EQ(18, decimal._exponent);
+    EXPECT_EQ(0, decimal._isNegative);
+
+    EXPECT_ARRAY_EQUAL(mantissa, decimal._mantissa);
+
+    //
+    dNumber = [[[NSDecimalNumber alloc] initWithDouble:9223372534876e-24] autorelease];
+    decimal = [dNumber decimalValue];
+    EXPECT_EQ(4, decimal._length);
+    EXPECT_EQ(-30, decimal._exponent);
+    EXPECT_EQ(0, decimal._isNegative);
+
+    EXPECT_ARRAY_EQUAL(mantissa, decimal._mantissa);
+
+    //
+    unsigned short ma[] = { 63488, 50716, 31416, 35167, 0, 0, 0, 0, 0 };
+    dNumber = [[[NSDecimalNumber alloc] initWithDouble:989876544e-6] autorelease];
+    decimal = [dNumber decimalValue];
+
+    EXPECT_EQ(4, decimal._length);
+    EXPECT_EQ(-16, decimal._exponent);
+    EXPECT_EQ(0, decimal._isNegative);
+
+    EXPECT_ARRAY_EQUAL(ma, decimal._mantissa);
+}
+
+TEST(NSDecimalNumber, DecimalNotANumber) {
+    NSDecimalNumber* dNumber = [[[NSDecimalNumber alloc] initWithMantissa:0 exponent:0 isNegative:YES] autorelease];
+    NSDecimal decimal = [dNumber decimalValue];
+
+    EXPECT_TRUE(NSDecimalIsNotANumber(&decimal));
+}
+
+TEST(NSDecimalNumber, DecimalCompact) {
+    NSDecimalNumber* dNumber = [[[NSDecimalNumber alloc] initWithLongLong:10000] autorelease];
+    NSDecimal decimal = [dNumber decimalValue];
+    NSDecimalCompact(&decimal);
+
+    EXPECT_EQ(1, decimal._length);
+    EXPECT_EQ(4, decimal._exponent);
+    EXPECT_EQ(0, decimal._isNegative);
+}
+
+TEST(NSDecimalNumber, Subtraction1) {
+    unsigned short mantissa[NSDecimalMaxSize] = { 24475, 10763, 48184, 32, 0, 0, 0, 0 };
+    NSDecimalNumber* dNumber = [[[NSDecimalNumber alloc] initWithUnsignedLongLong:9223372534876] autorelease];
+    NSDecimal left = [dNumber decimalValue];
+
+    dNumber = [[[NSDecimalNumber alloc] initWithLongLong:9223372036854775] autorelease];
+
+    NSDecimal right = [dNumber decimalValue];
+
+    NSDecimal result = {};
+
+    NSDecimalSubtract(&result, &left, &right, NSRoundPlain);
+    EXPECT_ARRAY_EQUAL(mantissa, result._mantissa);
+    EXPECT_EQ(4, result._length);
+    EXPECT_EQ(1, result._isNegative);
+    EXPECT_EQ(0, result._exponent);
+
+    NSDecimalSubtract(&result, &right, &left, NSRoundPlain);
```

Comment:
ooks like it disappeared. 

---
In reply to: [81399065](https://github.com/Microsoft/WinObjC/pull/1032#discussion_r81399065) [](ancestors = 81399065)